Example/Tests/CISAdAnalyticsTestDouble.swift

```diff
@@ -9,48 +9,88 @@
 import ConvivaSDK
 import Foundation
 
-class CISAdAnalyticsTestDouble: NSObject, CISAdAnalyticsProtocol {
+class CISAdAnalyticsTestDouble: NSObject, CISAdAnalyticsProtocol, TestDoubleDataSource {
     func isAirPlaying() -> Bool {
         false
     }
 
     func setAdInfo(_ adInfo: [AnyHashable: Any]) {
+        var args = [String: String]()
+        adInfo.forEach { key, value in args[String(describing: key)] = String(describing: value) }
+        spy(functionName: "setAdInfo", args: args)
```

Comment:
Why do you need to convert the input dict into a `String: String` dict here? 🤔 It's been a long time since I wrote this but IIRC the idea was to just pass in the value as you got it here.

What you do below for example in `reportAdFailed` is that you flatten the `errorMessage` + the `adInfo` dict together in one level dict. While there is no problem doing this, I think it was not the idea of this design 🤔 

```suggestion
        spy(
            functionName: "setAdInfo", 
            args: [
                "adInfo": String(describing: adInfo)
            ]
        )
```

This should provide everything you need for verifying certain calls 🤔

Reply:
I was trying that initially, but I think the order of the entries is not stable in the resulting string.
Also the resulting string looked very cumbersome. Something like `[AnyHashable(\"key\"): "value", AnyHashable(...`

IMO tests are a lot easier to read (and write) by flattening the map.

Reply:
As discussed. Let's create a pattern that could be also used in case of multiple dictionaries in the arguments (and that would not result in colliding keys then)

https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/69/commits/c62416281bf57b807e54259b865f78982a1c1ab6